### PR TITLE
warnings: add custom diff warnings support (bug 1699137)

### PR DESCRIPTION
**Pull request by zzzeid**
- add `DiffWarning` model
- add API endpoints and definitions
- add `RevisionWarningCheck` that fetches custom warnings
- add tests

TODOS:
- [x] add phabricator auth token requirement

### CODE REVIEW

File: landoapi/api/diff_warnings.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+"""
+This module provides the API controllers for the `DiffWarning` model.
+
+These API endpoints can be used by client (such as Lando UI, Code Review bot, etc.) to
```
Comment (zzzeid) [inline landoapi/api/diff_warnings.py:8]:
```suggestion
These API endpoints can be used by clients (such as Lando UI, Code Review bot, etc.) to
```

File: landoapi/api/diff_warnings.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def post(data):
+    """Create a new `DiffWarning` based on provided revision and diff IDs."""
+    # TODO: validate whether revision/diff exist or not.
+    # NOTE: should we be more specific about the type of data passed here?
+    warning = DiffWarning(**data)
+    db.session.add(warning)
+    db.session.commit()
+    return {"id": warning.id}, 201
```
Comment (zzzeid) [inline landoapi/api/diff_warnings.py:27]:
```suggestion
    return warning.serialize(), 201
```

File: landoapi/api/diff_warnings.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+
+def delete(pk):
+    """Archive a `DiffWarning` based on provided pk."""
+    warning = DiffWarning.query.get(pk)
+    warning.status = DiffWarningStatus.ARCHIVED
+    db.session.commit()
+    return {"id": pk}
```
Comment (zzzeid) [inline landoapi/api/diff_warnings.py:35]:
```suggestion
    return warning.serialize(), 200
```

**Review by dklawren — APPROVED** [review]
r=dkl

**Review by cgsheeh — APPROVED** [review]

**Pull request merged**